### PR TITLE
BUG: Frequency not set on empty series

### LINKED ISSUE #14320: BUG: Frequency not set for empty Series
Pandas doesn't set frequency properly for empty series. This is probably related to #14313.
#### A small, complete example of the issue

It works for non-empty series:

``` python
>>> pd.Series(index=pd.DatetimeIndex(["2016-09-29 11:00"]), data=[3]).asfreq('H').index
DatetimeIndex(['2016-09-29 11:00:00'], dtype='datetime64[ns]', freq='H')
```

But for empty series, `freq` is `None`:

``` python
>>> pd.Series(index=pd.DatetimeIndex([]), data=[]).asfreq('H').index
DatetimeIndex([], dtype='datetime64[ns]', freq=None)
```
#### Expected Output

I expected the following output:

``` python
>>> pd.Series(index=pd.DatetimeIndex([]), data=[]).asfreq('H').index
DatetimeIndex([], dtype='datetime64[ns]', freq='H')
```
#### Output of `pd.show_versions()`

<details>
## INSTALLED VERSIONS

commit: None
python: 3.5.2.final.0
python-bits: 64
OS: Linux
OS-release: 4.7.5-gnu-1
machine: x86_64
processor: 
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8

pandas: 0.19.0rc1
nose: 1.3.6
pip: 8.1.2
setuptools: 27.3.0
Cython: 0.24.1
numpy: 1.11.1
scipy: 0.18.1
statsmodels: None
xarray: None
IPython: 5.0.0
sphinx: 1.4.5
patsy: None
dateutil: 2.5.3
pytz: 2016.6.1
blosc: None
bottleneck: None
tables: 3.2.3.1
numexpr: 2.6.1
matplotlib: 1.5.2
openpyxl: None
xlrd: 1.0.0
xlwt: None
xlsxwriter: None
lxml: None
bs4: 4.5.1
html5lib: None
httplib2: 0.9.2
apiclient: 1.5.1
sqlalchemy: 1.0.15
pymysql: None
psycopg2: 2.6.2 (dt dec pq3 ext lo64)
jinja2: 2.8
boto: None
pandas_datareader: None
</details>

### PR DESCRIPTION
**Pull request by sahildua2305**
- [x] closes #14320 
- [x] tests added / passed
- [ ] passes `git diff upstream/master | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: pandas/tests/frame/test_timeseries.py [hunk 1]
```diff
@@ -323,6 +323,10 @@ def test_asfreq_datetimeindex(self):
         ts = df['A'].asfreq('B')
         tm.assertIsInstance(ts.index, DatetimeIndex)
 
+    def test_asfreq_datetimeindex_empty_series(self):
+        s = Series(index=pd.DatetimeIndex([]), data=[]).asfreq('H')
+        self.assertEqual(s.index.freq, 'H')
```
Comment (jreback) [inline pandas/tests/frame/test_timeseries.py]: construct an expected series and test against that

**Review by jreback — CHANGES_REQUESTED** [review]
add a release note in 0.20

File: pandas/tests/frame/test_timeseries.py (hunk 1 above)
Comment (sahildua2305) [inline pandas/tests/frame/test_timeseries.py]: Done! Please check.

File: doc/source/whatsnew/v0.20.0.txt [hunk 2]
```diff
@@ -79,3 +79,5 @@ Performance Improvements
 
 Bug Fixes
 ~~~~~~~~~
+
+- Bug in ``asfreq``, where frequency wasn't set for empty series (:issue:`14320`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.20.0.txt]: series -> ``Series``

File: pandas/tests/frame/test_timeseries.py [hunk 3]
```diff
@@ -323,6 +323,12 @@ def test_asfreq_datetimeindex(self):
         ts = df['A'].asfreq('B')
         tm.assertIsInstance(ts.index, DatetimeIndex)
 
+    def test_asfreq_datetimeindex_empty_series(self):
+        empty_series = Series(index=pd.DatetimeIndex([]), data=[]).asfreq('H')
```
Comment (jreback) [inline pandas/tests/frame/test_timeseries.py]: add the issue number as a comment

File: pandas/tests/frame/test_timeseries.py [hunk 4]
```diff
@@ -323,6 +323,12 @@ def test_asfreq_datetimeindex(self):
...
+        normal_series = Series(index=pd.DatetimeIndex(["2016-09-29 11:00"]),
```
Comment (jreback) [inline pandas/tests/frame/test_timeseries.py]: expected =

File: pandas/tests/frame/test_timeseries.py [hunk 5]
```diff
@@ -323,6 +323,12 @@ def test_asfreq_datetimeindex(self):
...
+                                                      data=[3]).asfreq('H')
+        self.assertEqual(empty_series.index.freq, normal_series.index.freq)
```
Comment (jreback) [inline pandas/tests/frame/test_timeseries.py]: use ``tm.assert_series_equal``

Comment (sahildua2305) [inline pandas/tests/frame/test_timeseries.py]: @jreback But we actually want to compare the frequency of the index not the complete series. Also, if we have one empty and one non-empty series, how can we use `assert_series_equal` method?

**Comment by jreback** [comment]
@sahildua2305 you need to rebase on master

``git rebase -i origin/master``

**Comment by sahildua2305** [comment]
Oops! Last time I rebased this branch it was fine. I will do that again.

**Comment by jreback** [comment]
if that doesn't work, then close and open a new PR. We had shifted the url of the repo.

**Comment by sahildua2305** [comment]
Sure!

**Pull request closed without merging**